### ISSUE TITLE
Library path/linking issue with Portaudio on macOS with CMake 3.18

### ISSUE THREAD

**Issue by dyfer** [supercollider/supercollider#5176]
## Environment

* SuperCollider version: develop
* Operating system: macOS 10.13
* Other details (Qt version, audio driver, etc.): CMake 3.18

## Steps to reproduce

Upgrade CMake. Configure and build.

## Expected vs. actual behavior

On macos we use Portaudio with Supernova.
During configuring, when `find_package(Portaudio REQUIRED)` is [called](https://github.com/supercollider/supercollider/blob/develop/server/supernova/CMakeLists.txt#L154), we get following output from CMake:

```
CMake Warning (dev) at /usr/local/Cellar/cmake/3.18.3/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:273 (message):
  The package name passed to `find_package_handle_standard_args` (PkgConfig)
  does not match the name of the calling package (Portaudio).  This can lead
  to problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.18.3/share/cmake/Modules/FindPkgConfig.cmake:59 (find_package_handle_standard_args)
  cmake_modules/FindPortaudio.cmake:22 (include)
  server/supernova/CMakeLists.txt:155 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found PkgConfig: /usr/local/bin/pkg-config (found version "0.29.2")
```

Inside [FindPortaudio.cmake](https://github.com/supercollider/supercollider/blob/develop/cmake_modules/FindPortaudio.cmake), the `PORTAUDIO2_LIBRARIES` variable seems initially to be set to a list: `portaudio;-framework CoreAudio;-framework AudioToolbox;-framework AudioUnit;-framework Carbon`, and then the resulting `PORTAUDIO_LIBRARIES` becomes 
```
/usr/local/Cellar/portaudio/19.6.0/lib/libportaudio;-framework CoreAudio;-framework AudioToolbox;-framework AudioUnit;-framework Carbon.dylib
```
It seems that we should only be taking the first element of that list. The correct paht should be `/usr/local/Cellar/portaudio/19.6.0/lib/libportaudio.dylib`. Unfortunately I can't easily revert CMake version to compare the behavior with the previous version.

This causes a linking error when compiling supernova:
```
clang: error: no such file or directory: '/usr/local/Cellar/portaudio/19.6.0/lib/libportaudio'
```
because the path is incorrect.

I created a workaround that sets seem to set the path correctly: https://github.com/dyfer/supercollider/commit/8e3d71beec0881a945e9336aab24085b2a8bd246 However, I'm not sure how to avoid the warning from CMake and what is the root cause of the problem.

**Comment by brianlheim** [comment]
the dev warning is spurious; it can be ignored at least for the purpose of this ticket.

probably we should not be using XYZ_LIBRARIES at all, this pkgconfig call sets many variables besides it: https://cmake.org/cmake/help/latest/module/FindPkgConfig.html

i will experiment with other variables and see if there is something more appropriate

**Comment by brianlheim** [comment]
@dyfer i think #5206 fixes this, give it a try if you have time

**Comment by dyfer** [comment]
Fixed in #5206 
Thanks @brianlheim !
